architecture/verifier/README.md

```diff
@@ -0,0 +1,86 @@
+# Verifier Refactor Changes Overview
+
+## Configuration
+
+Configuration for each Veraison component is encapsulated by a `ParamStore`
+that contains parameter definitions. A parameter definition specifies the type
+of the parameter (used in validation), and a path where it can be found within
+configuration.
+
+Configuration for a Veraison deployment is handled by a `Config` that gathers
+`ParamStore`s for components used in that deployment. The `Config`uses
+[Viper](https://github.com/spf13/viper) to gather configuration from various
+sources and populate `ParamStore`s associated with it. Parameters are populated
+based on their paths,  and the same value can be used to populate parameters
+inside multiple stores, if they share a path.
+
+<p/>
+<p align="center">
+  <img src="./veraison-config.png"/>
+</p>
+
+`ParamStore`s are created and initialized with parameter definitions for their
+respective component. Most components will have associated helper functions to
+do that. The `ParamStore`s are then passed onto the `Config` that assumes
+ownership of them, and populates them with values read from external sources.
+The values are validated against the definitions. This happens on deployment
+start. Once the `Config` finishes processing external sources, deployment
+initialization proceeds by initializing the individual components. The
+`ParamStore`s are passed to the components' `Init()` methods, whereupon the
+`Config` relinquishes ownership over them (i.e. they would no longer be
+updated). The values in the stores are used by the `Init()` methods to
+initialize the components. `ParamStores` are not used beyond that.
```

Comment:
On line 32 we have used the term `ParamStores` but I admit it does not look bad either ....